inference-engine/src/vpu/common/src/ngraph/transformations/dynamic_to_static_shape_binary_elementwise.cpp

```diff
@@ -48,9 +49,51 @@ void processBinaryEltwise(std::shared_ptr<ngraph::Node> eltwise, size_t lhsIndex
         broadcastInput = std::make_shared<ngraph::opset3::Concat>(ngraph::OutputVector{broadcastConst, broadcastInput}, 0);
     }
 
-    const auto shape = std::make_shared<ngraph::opset3::Maximum>(lhsInput, rhsInput);
+    const auto& lhsInputShape = lhsInput.get_partial_shape();
+    const auto& rhsInputShape = rhsInput.get_partial_shape();
 
-    auto outDSR = std::make_shared<ngraph::vpu::op::DynamicShapeResolver>(copied, shape);
+    VPU_THROW_UNLESS(lhsInputShape.is_static() && lhsInputShape.rank().get_length() == 1,
+        "DynamicToStaticShape transformation for {} of type {} expects lhs input shape to be static 1D vector, actual is {}",
+        eltwise->get_friendly_name(), eltwise->get_type_info(), lhsInputShape);
+
+    VPU_THROW_UNLESS(rhsInputShape.is_static() && rhsInputShape.rank().get_length() == 1,
+        "DynamicToStaticShape transformation for {} of type {} expects rhs input shape to be static 1D vector, actual is {}",
+        eltwise->get_friendly_name(), eltwise->get_type_info(), rhsInputShape);
+
+    std::shared_ptr<ngraph::Node> maxShape = std::make_shared<ngraph::opset3::Maximum>(lhsInput, rhsInput);
+    const auto& maxShapePartialShape = maxShape->output(0).get_partial_shape();
+    VPU_THROW_UNLESS(maxShapePartialShape.is_static(),
+        "DynamicToStaticShape transformation for {} of type {} expects output shape to have static rank, actual is dynamic");
+    const auto& maxShapeRank = ngraph::shape_size(maxShapePartialShape.to_shape());
+
+    const auto& updateOutputShapeOnZerosFrom = [&maxShape, &maxShapeRank, &shapeElementType](const ngraph::Output<ngraph::Node>& input) {
```

Comment:
Just to clarify, if we assume that we don't have negative shape values (as I can see from the description), should this code also work?
```
const auto& minimum = std::make_shared<ngraph::opset6::Minimum>(lhsInput, rhsInput);
outShape = std::make_shared<ngraph::opset6::Select>(
    std::make_shared<ngraph::opset6::Convert>(minimum, ngraph::element::boolean),
    maxShape,
    zeros);
```

Or there are some corner cases which I did not consider?

Reply:
Looks like this should work, but uses more operations:
Constant + Minimum + Select + Convert vs Constant + Equal + Select